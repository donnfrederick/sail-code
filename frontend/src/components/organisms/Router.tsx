import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
