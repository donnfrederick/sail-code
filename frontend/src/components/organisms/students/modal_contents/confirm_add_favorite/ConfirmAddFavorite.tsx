import Button from 'components/atoms/students/Button'
import ModalTextContainer from 'components/molecules/students/ModalTextContainer'
import CompleteAddFavoriteModal from 'components/organisms/students/modal_contents/complete_add_favorite'
import FailFetchingModal from 'components/organisms/students/modal_contents/fail_fetching'
import * as FavoritesModels from 'models/favorites'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  addFavoritedUserId: number
  authToken: string
  close(): void
  postFavorites(
    authToken: string,
    parameters: FavoritesModels.PostFavoritesRequest
  ): Promise<void>
  setContents(contents: JSX.Element): void
}

export default (props: Props) => {
  const {
    addFavoritedUserId,
    authToken,
    close,
    postFavorites,
    setContents
  } = props

  return (
    <Container>
      <ModalTextContainer heading={'Add this user?'} />
      <Text>
        {"This user's information will be added to your favorite users' list."}
      </Text>
      <ButtonContainer>
        <Button
          type="red"
          text="Cancel"
          width={272}
          height={112}
          fontSize={40}
          onClick={() => close()}
        />
        <Button
          type="white"
          text="OK"
          width={272}
          height={112}
          fontSize={40}
          onClick={async () => {
            if (addFavoritedUserId) {
              const parameters: FavoritesModels.PostFavoritesRequest = {
                user_id: addFavoritedUserId
              }

              try {
                await postFavorites(authToken, parameters)
                setContents(<CompleteAddFavoriteModal />)
              } catch (error) {
                // tslint:disable-next-line
                console.error(error)
                setContents(<FailFetchingModal />)
              }
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

const Text = styled.div`
  width: 530px;
  margin: 0 auto 50px;
  padding-left: 1em;
  font-size: 24px;
  line-height: 1.38;
  color: #405766;
  text-align: left;
  text-indent: -1em;
`
