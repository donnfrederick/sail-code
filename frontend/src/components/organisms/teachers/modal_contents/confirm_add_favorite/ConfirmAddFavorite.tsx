import Button from 'components/atoms/teachers/Button'
import ModalTextContainer from 'components/molecules/teachers/ModalTextContainer'
import CompleteAddFavoriteModal from 'components/organisms/teachers/modal_contents/complete_add_favorite'
import FailFetchingModal from 'components/organisms/teachers/modal_contents/fail_fetching'
import * as FavoritesModels from 'models/favorites'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  authToken: string
  addFavoritedUserId: number
  close(): void
  postFavorites(
    authToken: string,
    parameters: FavoritesModels.PostFavoritesRequest
  ): Promise<void>
  setContents(contents: JSX.Element): void
}

export default (props: Props) => {
  const {
    authToken,
    addFavoritedUserId,
    close,
    postFavorites,
    setContents
  } = props

  return (
    <Container>
      <ModalTextContainer heading={'この学生をお気に入りに追加しますか？'} />
      <Text>
        {
          'お気に入りに追加した学生は、[メニュー] > [お気に入り一覧] から確認することができます。'
        }
      </Text>
      <ButtonContainer>
        <Button
          type="white"
          text="いいえ"
          width={272}
          height={112}
          fontSize={40}
          onClick={() => close()}
        />
        <Button
          type="blue"
          text="はい"
          width={272}
          height={112}
          fontSize={40}
          onClick={async () => {
            if (addFavoritedUserId) {
              const parameters: FavoritesModels.PostFavoritesRequest = {
                user_id: addFavoritedUserId
              }

              try {
                await postFavorites(authToken, parameters)
                setContents(<CompleteAddFavoriteModal />)
              } catch (error) {
                // tslint:disable-next-line
                console.error(error)
                setContents(<FailFetchingModal />)
              }
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

const Text = styled.div`
  width: 530px;
  margin: 0 auto 50px;
  padding-left: 1em;
  font-size: 24px;
  line-height: 1.38;
  color: #405766;
  text-align: left;
  text-indent: -1em;
`
