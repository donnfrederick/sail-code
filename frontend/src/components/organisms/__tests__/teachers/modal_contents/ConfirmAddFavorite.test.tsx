import Router from 'components/organisms/Router'
import ConfirmAddFavorite from 'components/organisms/teachers/modal_contents/confirm_add_favorite'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

test('render <ConfirmBlock />', () => {
  const tree = renderer
    .create(
      <Router>
        <ConfirmAddFavorite addFavoriteUserId={1} />
      </Router>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
