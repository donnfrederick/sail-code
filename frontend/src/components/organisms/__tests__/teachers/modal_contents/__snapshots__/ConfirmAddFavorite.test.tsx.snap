// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <ConfirmBlock /> 1`] = `
<div
  className="sc-kAzzGY gEvfts"
>
  <div
    className="sc-VigVT kaYUeB"
  >
    <div
      className="sc-gZMcBi fxOrEW"
    >
      この学生をお気に入りに追加しますか？
    </div>
  </div>
  <div
    className="sc-kgoBCf jyrmPB"
  >
    お気に入りに追加した学生は、[メニュー] &gt; [お気に入り一覧] から確認することができます。
  </div>
  <div
    className="sc-chPdSV hiSuyO"
  >
    <button
      className="sc-EHOje dyvxUy"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      width={272}
    >
      <div
        className="sc-iwsKbI lgKvmV"
      >
        いいえ
      </div>
    </button>
    <button
      className="sc-bZQynM iiToMd"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      width={272}
    >
      <div
        className="sc-iwsKbI lgKvmV"
      >
        はい
      </div>
    </button>
  </div>
</div>
`;
